tests/tpch/test_optimization.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+
+from . import dask_queries
+
+pytestmark = pytest.mark.tpch_dask
+
+
+@pytest.mark.parametrize(
+    "query",
+    [
+        1,
+        2,
+        3,
+        4,
+        5,
+        6,
+        7,
+        8,
+        9,
+        10,
+        11,
+        12,
+        13,
+        14,
+        15,
+        16,
+        17,
+        18,
+        19,
+        20,
+        21,
+        22,
+    ],
+)
+def test_optimization(query, dataset_path, fs, client):
+    func = getattr(dask_queries, f"query_{query}")
+    result = func(dataset_path, fs)
+    # We need to inject .repartition(npartitions=1) which .compute() does under the hood
+    result.repartition(npartitions=1).optimize()
```

Comment:
by default, this is using a coiled Cluster. Is this intentional? This makes running this test in isolation really expensive

Reply:
> by default, this is using a coiled Cluster. Is this intentional? This makes running this test in isolation really expensive

This strictly follows the deployment pattern of the actual TPC-H benchmarks to account for necessary pre-computations. This allows us to match the measured performance to the numbers we retrieve for the total computation. You should be able to run this in isolation with `--local` without any Coiled infrastructure. However, this would then also read local data which will skew the measurements (but should be fine for CPU-based stuff).

Reply:
> Benchmark results look odd given that Q18 takes 22 - 25 seconds to complete and 80 seconds to optimize.
> 
> <img alt="Screenshot 2024-02-01 at 10 44 31" width="916" src="https://private-user-images.githubusercontent.com/2699097/301459170-8fb9ec66-6b11-4ea2-b3c2-ed067580a052.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3OTUzMzgsIm5iZiI6MTcwNjc5NTAzOCwicGF0aCI6Ii8yNjk5MDk3LzMwMTQ1OTE3MC04ZmI5ZWM2Ni02YjExLTRlYTItYjNjMi1lZDA2NzU4MGEwNTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMDFUMTM0MzU4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmFkMjkwNWZiYjg0ZGQwOTdlMDk4ZjU1YzEwYWU3YWY4YWI3ZTQ1MWNmZWI2ZmFmYzc1MDViY2Q5Nzk2ZTViNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.9I6O6gv7nSM31Ys9TaIx2TWnCt9-yD2Q8BrJqIDAM3U">
> EDIT: This may have been caused by me using a `LocalCluster`, the query sets an index and requires some pre-computation.

This shows what happens when running this without a Coiled cluster but using cloud-based data. For now, I'm inclined to keep it this way and investigate the benchmarked queries first. There are already suggestions on how to reduce the amount of pre-computation, so we adjusting this might introduce less skew in the future.